### PR TITLE
f-button@5.0.0 - Split small button into productive and expressive

### PR DESCRIPTION
**Pull request by xander-marjoram**
### Removed
- `small` button size (this still remains for icon buttons)

### Added
- `small-productive` button size (not for icon buttons)
- `small-expressive` button size (not for icon buttons)
  - These buttons are the same height but expressive has a larger font and therefore less vertical padding.

### Changed
- Use CSS logical properties for padding, margins, heights, widths and spinner positioning.

---

And various linting fixes for some other components

---

## UI Review Checks

- [x] README and/or UI Documentation has been [created|updated]
- [x] Unit tests have been [created|updated]
- [ ] This code has been checked with regard to [our accessibility standards](http://fozzie.just-eat.com/documentation/general/accessibility/checklist)

## Browsers Tested

- [x] Chrome (latest)
- [ ] Internet Explorer 11
- [ ] Mobile (Please list device/browser – Ideally one iPhone model and one Android)

### List any other browsers that this PR has been tested in:

- [View the Browser Support Checklist](http://fozzie.just-eat.com/documentation/general/browser-support)

### CODE REVIEW

File: packages/components/atoms/f-button/stories/Button.stories.js [hunk 1]
```diff
@@ -56,28 +56,32 @@ ButtonComponent.argTypes = {
 ButtonComponent.args = {
     buttonLists: [
         [
-            { type: 'primary', size: 'large', text: 'Large Primary Button' },
-            { type: 'primary', size: 'medium', text: 'Medium Primary Button' },
-            { type: 'primary', size: 'small', text: 'Small Primary Button' },
-            { type: 'primary', size: 'xsmall', text: 'Xsmall Primary Button' }
+            { type: 'primary', size: 'large', text: 'Large Primary' },
```
Comment (xander-marjoram) [inline packages/components/atoms/f-button/stories/Button.stories.js:59]: Got rid of the superfluous word "Button" on each of these to free up a bit of space.

File: packages/components/pages/f-checkout/stories/api/responses/getBasket/getBasket.js [hunk 2]
```diff
@@ -9,33 +9,31 @@ const restriction = {
     Type: 'Alcohol'
 };
 
-function buildPrompts (issues) {
-    return {
-        SpendMore: null,
-        DiscountApplied: {
-            Amount: 0
+const buildPrompts = issues => ({
```
Comment (jamieomaguire) [inline packages/components/pages/f-checkout/stories/api/responses/getBasket/getBasket.js:12]: this is fine, but I think next time it might be simpler to raise a separate PR straight after that contains purely linting fixes 😃

**Review by jamieomaguire — APPROVED** [review]

**Review by LTurns — APPROVED** [review]
